planning/behavior_path_planner/src/utils/avoidance/utils.cpp

```diff
@@ -675,21 +675,62 @@ void filterTargetObjects(
       o.overhang_lanelet = overhang_lanelet;
       lanelet::BasicPoint3d overhang_basic_pose(
         o.overhang_pose.position.x, o.overhang_pose.position.y, o.overhang_pose.position.z);
+
       const bool get_left = isOnRight(o) && parameters->enable_avoidance_over_same_direction;
       const bool get_right = !isOnRight(o) && parameters->enable_avoidance_over_same_direction;
+      const bool get_opposite = parameters->enable_avoidance_over_opposite_direction;
 
-      const auto target_lines = rh->getFurthestLinestring(
-        overhang_lanelet, get_right, get_left,
-        parameters->enable_avoidance_over_opposite_direction);
+      const auto merge_two_line = [](const auto & a, const auto & b) {
+        lanelet::Points3d new_points;
+        for (const auto & p : a) {
+          new_points.emplace_back(lanelet::utils::getId(), p.basicPoint());
+        }
 
-      if (isOnRight(o)) {
-        o.to_road_shoulder_distance =
-          distance2d(to2D(overhang_basic_pose), to2D(target_lines.back().basicLineString()));
-        debug.bounds.push_back(target_lines.back());
-      } else {
-        o.to_road_shoulder_distance =
-          distance2d(to2D(overhang_basic_pose), to2D(target_lines.front().basicLineString()));
-        debug.bounds.push_back(target_lines.front());
+        for (const auto & p : b) {
+          new_points.emplace_back(lanelet::utils::getId(), p.basicPoint());
+        }
+        return lanelet::LineString3d(lanelet::utils::getId(), new_points);
+      };
+
+      lanelet::ConstLineString3d target_line{};
+
+      lanelet::ConstLanelets previous_lanelet{};
+      if (rh->getPreviousLaneletsWithinRoute(overhang_lanelet, &previous_lanelet)) {
+        const auto lines =
+          rh->getFurthestLinestring(previous_lanelet.front(), get_right, get_left, get_opposite);
+        if (isOnRight(o)) {
+          target_line = merge_two_line(target_line, lines.back());
+        } else {
+          target_line = merge_two_line(target_line, lines.front());
+        }
```

Comment:
memo:
getFurthestLinestring() push linestring in the order of right -> left. so the target object is on the right side,need to get left line which lines.back().
https://github.com/autowarefoundation/autoware.universe/blob/main/planning/route_handler/src/route_handler.cpp#L1166